### PR TITLE
replace math/rand with math/rand/v2

### LINKED ISSUE #5335: Use math/rand/v2 in place of math/rand
**Describe the enhancement:**

In Go 1.22 we got a new package math/rand/v2 that replaces the aging math/rand.

Some improvements include better performance and scalability as well as safer algorithms. See [Evolving the Go Standard Library with math/rand/v2](https://go.dev/blog/randv2).

Since our go.mod file requires go1.22.4 minimum we can safely depend on this new package.

**What is the definition of done?**

All occurrences of math/rand are replaced with math/rand/v2.

Grepping for math/rand usage I got:

```go
$ rg -l '"math/rand"' | xargs rg 'rand\.'
main.go
39:     rand.Seed(time.Now().UnixNano())

internal/pkg/scheduler/scheduler.go
133:    return time.Duration(rand.Int63n(t))

magefile.go
2109:           bkpName := fmt.Sprintf("./env.sh-%d", rand.Int())

internal/pkg/remote/client.go
267:    rand.Shuffle(len(clients), func(i, j int) {

internal/pkg/core/backoff/equal_jitter.go
33:             nextRand: time.Duration(rand.Int63n(int64(init))), //nolint:gosec
54:     b.nextRand = time.Duration(rand.Int63n(int64(b.duration)))

testing/integration/install_test.go
337:    rand.Seed(time.Now().UnixNano())
342:            runes[i] = letters[rand.Intn(len(letters))]

internal/pkg/agent/cmd/enroll_cmd.go
707:    t := time.NewTimer(time.Duration(rand.Int63n(int64(d))))

internal/pkg/agent/application/upgrade/marker_access_test.go
162:            rune := chars[rand.Intn(len(chars))]

internal/pkg/agent/application/upgrade/artifact/download/http/verifier_test.go
114:    first := rand.Intn(len(tt))
115:    second := rand.Intn(len(tt))
```

- `rand.Seed(time.Now().UnixNano())` rand.Seed is deprecated since 1.20 and removed in v2, see https://github.com/elastic/elastic-agent/pull/5334
- `rand.Int` stays in v2, no changes needed.
- `rand.Intn` stays in v2 as `rand.IntN`
- `rand.Int63n(x)` can now be written as `rand.N(x)` as stated in the blog post.
- `rand.Shuffle` stays the same and gains some perf improvements as stated in the blog post.

### PR DESCRIPTION
**Pull request by mauri870**
## What does this PR do?

This PR updates all occurrences of math/rand with the new math/rand/v2 introduced in Go 1.22.

## Why is it important?

The math/rand/v2 includes performance and scalability improvements.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

- Closes #5335

### CODE REVIEW

**Review by blakerouse — APPROVED** [review]
Very mechanical, looks good.

File: changelog/fragments/1724253791-replace-math-rand-with-math-rand-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Kind can be one of:
```
Comment (mauri870) [inline changelog/fragments/1724253791-replace-math-rand-with-math-rand-v2.yaml:1]: I'm not completely sure if this needs a changelog. I saw that @pierrehilbert added a skip-changelog tag, that means I can remove this fragment or is it better to keep it?

Comment (jlind23) [reply changelog/fragments/1724253791-replace-math-rand-with-math-rand-v2.yaml:1]: It means you can remove it.

Comment (mauri870) [reply changelog/fragments/1724253791-replace-math-rand-with-math-rand-v2.yaml]: Thanks!

**Comment by mauri870** [comment]
Test failure seems to be unrelated and is being tracked at https://github.com/elastic/elastic-agent/issues/5279.

**Comment by VihasMakwana** [comment]
@mauri870 I've merged main into this. https://github.com/elastic/elastic-agent/issues/5279 is now resolved and CI should be green.

**Comment by mauri870** [comment]
> @mauri870 I've merged main into this. #5279 is now resolved and CI should be green.

Thanks!

**Review by pchila — APPROVED** [review]
LGTM

**Pull request merged**